src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1778,7 +1778,7 @@ public void MainWindowTitle_GetWithGui_ShouldRefresh_Windows()
             }
         }
 
-        [Fact]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotBuiltWithAggressiveTrimming))]
```

Comment:
```suggestion
        [Fact]
        [DynamicDependency(DynamicallyAccessedMemberTypes.NonPublicFields, typeof(Process))]
```

Reply:
This doesn't work. `error IL2037: System.Diagnostics.Tests.ProcessTests.RefreshResetsAllRefreshableFields(): No members were resolved for 'NonPublicFields'.` on WASM legs because of WarnAsError and illink trying to be "helpful".

NativeAOT is also still crashing with a NullRef.

I don't think there's enough value in this test to spend this much time on it trying to make it work.

Reply:
> NativeAOT is also still crashing with a NullRef.

Why is that? Is it a bug or expected?

I agree that the test itself is not super valuable for AOT. However, I believe that there is a value in trying to use the annotations in cases like these to validate that they work as expected and to tech people how to annotate the private reflection patterns correctly.

Reply:
> Why is that? Is it a bug or expected?

Likely because of the reflection on `System.Diagnostics.ProcessInfo` below.

> tech people how to annotate the private reflection patterns correctly.

There are only two places where DynamicDependency is an appropriate things to use:

* Internal reflection happening within the runtime (pretty much just corelib is eligible), or
* Other reflection happening outside trimming reach (pretty much just hosting scenarios).

It's not the appropriate thing to use anywhere else because it doesn't make the analysis warning go away. One needs to suppress the warnings with DynamicDependency and most people will get the suppressions wrong.

I wouldn't be able to rewrite this to use the correct pattern because uses of `object.GetType` can only be correctly done by placing `DynamicallyAccessedMembers` on the type itself, or making it sealed. But we don't want to mark Process as DynamicallyAccessedMembers or mark it sealed just so that the test can get to the internals.

Reply:
This is weird:
```C#
typeof(Process).GetField(fieldName, BindingFlags.NonPublic | BindingFlags.Instance)
```

Is basically going to do the same thing as 
```C#
[DynamicDependency(DynamicallyAccessedMemberTypes.NonPublicFields, typeof(Process))]
```

So we should not need the DynamicDependency.

And we could rewrite:
```C#
typeof(Process).Assembly.GetType("System.Diagnostics.ProcessInfo")
```

To
```C#
Type.GetType("System.Diagnostics.ProcessInfo")
```

and that part should also work.